root/types.js

```diff
@@ -1,24 +1,72 @@
-// @flow
-// Copyright (C) 2017 MetaBrainz Foundation
-//
-// This file is part of MusicBrainz, the open internet music database,
-// and is licensed under the GPL version 2, or (at your option) any
-// later version: http://www.gnu.org/licenses/gpl-2.0.txt
+/*
+ * @flow
+ * Copyright (C) 2017 MetaBrainz Foundation
+ *
+ * This file is part of MusicBrainz, the open internet music database,
+ * and is licensed under the GPL version 2, or (at your option) any
+ * later version: http://www.gnu.org/licenses/gpl-2.0.txt
+ */
 
 
-// Types are in alphabetical order.
-//
-// The definitions in this file are intended to model the output of the
-// TO_JSON methods under lib/MusicBrainz/Server/Entity/; those are precisely
-// how data is serialized for us.
+/*
+ * Types are in alphabetical order.
+ *
+ * The definitions in this file are intended to model the output of the
+ * TO_JSON methods under lib/MusicBrainz/Server/Entity/, those are precisely
+ * how data is serialized for us.
+ */
+
+declare type AreaT =
+  & CoreEntityT
+  & TypeRoleT<AreaTypeT>
```

Comment:
Added a `TypeRoleT` to model `MusicBrainz::Server::Entity::Role::Type`.

It doesn't currently use the type parameter, but I'm thinking we'll add a `type` prop there (to replace `typeName`, i.e. we'd access `entity.type.name` instead, and have access to the full type).